### ISSUE TITLE
NullnessChecker false positive when using Comparator#nulls{First,Last} 

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#402]
```
What steps will reproduce the problem?
Compile the attached Java class using `javac -processor 
org.checkerframework.checker.nullness.NullnessChecker -version -verbose 
-AprintErrorStack -AprintAllQualifiers NullableStringWrapper.java`

What is the expected output? What do you see instead?
I expect the class to compile successfully. Instead, the nullness analysis 
fails because the first arguments to Comparator#comparing and 
Comparator#thenComparing are functions that may return null. These errors are 
false positives because the second arguments to these methods are null-safe 
comparators.  

What version of the product are you using? On what operating system?
I built the "trunk" versions of jsr308-langtools, annotation-tools and 
checker-framework from source, as described in the manual, using jdk1.8.0_31. 
I'm running linux.

Please provide any additional information below.
I understand that the null-safety of the code depends crucially on the 
comparators used. In this case the code is safe, and it'd be great if the 
Checker Framework could detect that.

```

Original issue reported on code.google.com by `stephan...@gmail.com` on 28 Feb 2015 at 9:33

Attachments:
- [NullableStringWrapper.java](https://storage.googleapis.com/google-code-attachments/checker-framework/issue-402/comment-0/NullableStringWrapper.java)
- [NullableStringWrapper-compiler-output.txt](https://storage.googleapis.com/google-code-attachments/checker-framework/issue-402/comment-0/NullableStringWrapper-compiler-output.txt)

**Comment by smillst** [comment]
This false positives should go away once #979 is fixed. (These test cases will be added to that issue.)
